code-experiments/build/python/example_experiment_non_anytime.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import unicode_literals
+import cocoex  # experimentation module
+import numpy as np
+
+# for postprocessing
+import cocopp
+import os
+import webbrowser
+
+# for solvers
+import scipy.optimize
+import cma
+
+
+def default_budget_list(max_budget=10, num=100):
+    """Produces a budget list with at most `num` different increasing budgets
+    within [1, `max_budget`] that are equally spaced in the logarithmic space.
+    """
+    return np.unique(np.logspace(0, np.log10(max_budget), num=num).astype(int))
+
+
+def fmin(problem, x0, solver, budget):
+    """Invokes `solver` on `problem` with initial solution `x0`.
+
+    `solver` should evaluate the final/returned solution. Add here the
+    interface to your solver.
+    """
+    if solver.__name__ == 'fmin':
+
+        if solver.__globals__['__name__'] in ['cma', 'cma.evolution_strategy',
+                                              'cma.es']:
+            # set verbose=1 for more output
+            return solver(problem, x0, 2, options=dict(maxfevals=budget,
+                                                       verbose=-9))
+
+        elif solver.__globals__['__name__'] == 'scipy.optimize.optimize':
+            return solver(problem, x0, xtol=1e-9, ftol=1e-9, maxfun=budget,
+                          disp=False)  # set disp=True for more output
+
+        else:
+            return solver(problem, x0)
+
+    elif solver.__name__ == 'differential_evolution':
+        pop_size_multiplier = 5  # pop_size=pop_size_multiplier*dimension
+        # a strange but correct way to set max_iter to respect the budget
+        max_iter = max(1, int(
+            budget / pop_size_multiplier / problem.dimension) - 1)
+        bounds = [x for x in zip(problem.lower_bounds, problem.upper_bounds)]
+        return solver(problem, bounds, popsize=pop_size_multiplier,
+                      maxiter=max_iter, strategy='best1bin',
+                      tol=1e-9, polish=False,
+                      disp=False)  # set disp=True for more output
+
+    else:
+        return solver(problem, x0)
+
+
+def main():
+    ### input
+    suite_name = "bbob"
+    solver = scipy.optimize.fmin  # the Nelder-Mead downhill simplex method
+    # solver = scipy.optimize.differential_evolution  # to use DE instead
+    # solver = cma.fmin  # to use CMA-ES instead
+    algorithm_name = "Nelder-Mead"  # no spaces allowed
+    output_folder = algorithm_name  # no spaces allowed
+
+    # a list of increasing budgets (to be additionally multiplied by dimension)
+    # gradually increase `max_budget` to 10, 100, ...
+    # or replace with a user-defined list
+    budget_multiplier_list = default_budget_list(max_budget=5)
+
+    ### prepare (see http://numbbo.github.io/coco-doc/C/ for parameters)
+    suite = cocoex.Suite(suite_name, "", "")
+    observer = cocoex.Observer(suite_name,
+                               "result_folder: {} algorithm_name: {}".format(
```

Comment:
For the record, this syntax is not compatible with Python 2.6, where you would need to specify the argument ID with the `{}` brackets, like ``"result_folder: {0} algorithm_name: {1}".format(``.